src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1415,7 +1414,7 @@ void ProcessNewTransactionStatus(TransactionStatus newStatus)
 
         internal void ClearTransaction()
         {
-            Transaction.DisposeImmediately();
+            Transaction?.DisposeImmediately();
```

Comment:
Is there a valid runtime reason for ClearTransaction to be called with Transaction being null? If not, this should be a Debug.Assert

Reply:
There is - if transactions are managed by queries (begin/commit/rollback)

Reply:
Makes sense, thanks.